src/Avalonia.Controls/Notifications/WindowNotificationManager.cs

```diff
@@ -0,0 +1,149 @@
+            PseudoClass<WindowNotificationManager, NotificationPosition>(PositionProperty, x => x == NotificationPosition.BottomLeft, ":bottomleft");
+            PseudoClass<WindowNotificationManager, NotificationPosition>(PositionProperty, x => x == NotificationPosition.BottomRight, ":bottomright");
+
+            HorizontalAlignmentProperty.OverrideDefaultValue<WindowNotificationManager>(Layout.HorizontalAlignment.Stretch);
+            VerticalAlignmentProperty.OverrideDefaultValue<WindowNotificationManager>(Layout.VerticalAlignment.Stretch);
+        }
+
+        protected override void OnTemplateApplied(TemplateAppliedEventArgs e)
+        {
+            base.OnTemplateApplied(e);
+
+            var itemsControl = e.NameScope.Find<Panel>("PART_Items");
+            _items = itemsControl?.Children;
+        }
+
+        public void Show(INotification content)
+        {
+            Show(content as object);
+        }
+
+        public async void Show(object content)
+        {
+            var notification = content as INotification;
+
+            var notificationControl = new Notification
+            {
+                Content = content
+            };
+
+            if (notification != null)
+            {
+                notificationControl.NotificationClosed += (sender, args) => notification.OnClose?.Invoke();
+            }
+
+            notificationControl.NotificationClosed += OnNotificationClosed;
+
+            notificationControl.PointerPressed += (sender, args) =>
+            {
+                if (notification != null && notification.OnClick != null)
+                {
+                    notification.OnClick.Invoke();
+                    (sender as Notification)?.Close();
+                }
+            };
+
+            lock (_items)
+            {
+                _items.Add(notificationControl);
+
+                if (_items.OfType<Notification>().Count(i => !i.IsClosing) > MaxItems)
+                {
+                    _items.OfType<Notification>().First(i => !i.IsClosing).Close();
+                }
+            }
+
+            if (notification != null && notification.Expiration == TimeSpan.MaxValue)
+            {
+                return;
+            }
+
+            await Task.Delay(notification?.Expiration ?? TimeSpan.FromSeconds(5));
+
+            notificationControl.Close();
+        }
+
+        private void OnNotificationClosed(object sender, RoutedEventArgs routedEventArgs)
+        {
+            var notification = sender as Notification;
+            _items.Remove(notification);
+        }
+
+        public void Install(Window host)
+        {
+            var adornerLayer = host.GetVisualDescendants()
+                .OfType<AdornerDecorator>()
+                .FirstOrDefault()
+                ?.AdornerLayer;
+
+            if (adornerLayer != null)
+            {
+                adornerLayer.Children.Add(this as IControl);
```

Comment:
I don't think this cast is necessary.

```suggestion
                adornerLayer.Children.Add(this);
```

Reply:
resolved